### PR TITLE
Implemented showing artist details

### PR DESCRIPTION
**Pull request by poisdeux**
When user selects artist from ArtistListFragment it will now
show a screen displaying artist details. I've taken the TV show
details setup as an example.

This adds the following new functionality:

* Displaying artist fanart
* Displaying artist description
* Download all songs from an artist

In my opinion this makes the UX more consistent. Hope you like it as well.

### CODE REVIEW

File: app/src/main/java/org/xbmc/kore/provider/MediaProvider.java [hunk 1]
```diff
@@ -77,6 +77,7 @@
     private static final int ALBUM_GENRES_LIST = 711;
 
     private static final int SONGS_ALL = 800;
+    private static final int SONGS_ARTIST = 801;
     private static final int SONGS_LIST = 802;
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/provider/MediaProvider.java]: For coherence, this should be changed to SONGS_ALBUM

File: app/src/main/java/org/xbmc/kore/provider/MediaProvider.java [hunk 2]
```diff
@@ -663,6 +668,16 @@ private SelectionBuilder buildQuerySelection(Uri uri, int match) {
...
+            case SONGS_ARTIST: {
+                final String hostId = MediaContract.Hosts.getHostId(uri);
+                final String artistId = MediaContract.Artists.getArtistId(uri);
+                return builder.table(MediaDatabase.Tables.SONGS)
+                              .join(MediaDatabase.Tables.ALBUM_ARTISTS)
+                              .where(MediaDatabase.Tables.SONGS + "." + MediaContract.Songs.HOST_ID + "=?", hostId)
+                              .where(MediaDatabase.Tables.SONGS + "." + MediaContract.Songs.ALBUMID + "=" +
+                                     MediaDatabase.Tables.ALBUM_ARTISTS + "." + MediaContract.AlbumArtists.ALBUMID)
+                              .where(MediaContract.AlbumArtists.ARTISTID + "=?", artistId);
+            }
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/provider/MediaProvider.java]: This section doesn't follow the same pattern as the others. 
Take a look at the `ARTIST_ALBUMS_LIST` section for an example. You'll need to create a `MediaDatabase.Tables.SONGS_FOR_ARTIST_JOIN` (similar to `MediaDatabase.Tables.ALBUMS_FOR_ARTIST_JOIN`) that does the join and then use it in this section.
No need to introduce the `builder.join` method.

File: app/src/main/java/org/xbmc/kore/ui/ArtistDetailsFragment.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import org.xbmc.kore.utils.TabsAdapter;
+import org.xbmc.kore.utils.UIUtils;
+import org.xbmc.kore.utils.Utils;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+
+/**
+ * Container for the TV Show overview and Episodes list
+ */
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/ArtistDetailsFragment.java]: Update comment

File: app/src/main/java/org/xbmc/kore/ui/ArtistOverviewFragment.java [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import butterknife.OnClick;
+
+/**
+ * Presents movie details
+ */
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/ArtistOverviewFragment.java]: Comment

File: app/src/main/java/org/xbmc/kore/utils/SelectionBuilder.java [hunk 5]
```diff
@@ -83,6 +85,11 @@ public SelectionBuilder where(String selection, String... selectionArgs) {
         return this;
     }
 
+    public SelectionBuilder join(String table) {
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/utils/SelectionBuilder.java]: No need for this

**Comment by SyncedSynapse** [comment]
Have to go, more comments later...

File: app/src/main/java/org/xbmc/kore/ui/ArtistOverviewFragment.java [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+                     .show();
+            }
+        }, callbackHandler);
+    }
+
+    @OnClick(R.id.add_to_playlist)
+    public void onAddToPlaylistClicked(View v) {
+        final PlaylistType.Item playListItem = new PlaylistType.Item();
+        playListItem.artistid = artistId;
+        MediaPlayerUtils.queueAudio(this, playListItem);
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/ArtistOverviewFragment.java]: Breaks the build. You changed the signature of this method in your last PR, so this should reflect that.

**Comment by SyncedSynapse** [comment]
Looks good and is indeed better than it was.

My only question about this is that it introduces one more step to get to the artist's albums (one more click), which usually is what the user wants to do primarily. This could be fixed by making the "Albums" tab the default for the activity. Only downside is that we would loose the shared element animation, which i personally like a lot...

**Comment by poisdeux** [comment]
I think the extra effort to get to the artist albums is a minor problem, as I believe most users will use a touchscreen device and it will only take a simple swipe to get the list of albums. Personally, I usually want to listen to an artist and not a specific album. If I do want to listen to a specific album, I go to the album list and use search.
But the real reason I wanted this, is that in my opinion it enhances the UX of listening to music. Just like we used to do (long, long ago....) reading the booklet or vinyl cover :D

File: app/src/main/java/org/xbmc/kore/provider/MediaProvider.java (hunk 2 above)
Comment (poisdeux) [inline app/src/main/java/org/xbmc/kore/provider/MediaProvider.java]: Ah I didn't see that. Fixed now.

**Comment by poisdeux** [comment]
Code updated

File: app/src/main/java/org/xbmc/kore/provider/MediaDatabase.java [hunk 7]
```diff
@@ -88,7 +88,14 @@
...
                 " AND " +
                 ALBUM_GENRES + "." + MediaContract.AlbumGenres.GENREID + "=" + AUDIO_GENRES + "." + MediaContract.AudioGenres.GENREID;
-	}
+
+        /**
+         * Join to get Songs for an Artist
+         */
+        String SONGS_FOR_ARTIST_JOIN =
+                SONGS + " JOIN " + ALBUM_ARTISTS + " ON " +
+                SONGS + "." + MediaContract.Songs.ALBUMID + "=" + ALBUM_ARTISTS + "." + MediaContract.AlbumArtists.ALBUMID;
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/provider/MediaDatabase.java]: Add HOST_ID to the join to be unique.

Comment (poisdeux) [inline app/src/main/java/org/xbmc/kore/provider/MediaDatabase.java]: I thought it was redundant as host id was added to the where clause in the builder. But after taking a closer look I noticed it isn't the same. Sorry about that.

**Pull request merged**